tests/integration/test_spanner_graph_qa.py

```diff
@@ -0,0 +1,215 @@
+                    ),
+                    type="LivesIn",
+                    properties={},
+                ),
+                Relationship(
+                    source=Node(
+                        id="Elias Thorne", type="Person" + type_suffix, properties={}
+                    ),
+                    target=Node(
+                        id="Zephyr", type="Animal" + type_suffix, properties={}
+                    ),
+                    type="Owns",
+                    properties={},
+                ),
+                Relationship(
+                    source=Node(id="Elara", type="Person" + type_suffix, properties={}),
+                    target=Node(
+                        id="Capital City", type="Location" + type_suffix, properties={}
+                    ),
+                    type="LivesIn",
+                    properties={},
+                ),
+                Relationship(
+                    source=Node(
+                        id="Elias Thorne", type="Person" + type_suffix, properties={}
+                    ),
+                    target=Node(id="Elara", type="Person" + type_suffix, properties={}),
+                    type="Sibling",
+                    properties={},
+                ),
+            ],
+            source=Document(
+                metadata={},
+                page_content=(
+                    "Elias Thorne lived in the desert. He was a skilled craftsman"
+                    " who worked with sandstone. Elias had a pet falcon named"
+                    " Zephyr. His sister, Elara, resided in the capital city and"
+                    " ran a spice shop. They rarely met due to the distance."
+                ),
+            ),
+        )
+    ]
+    graph.add_graph_documents(graph_documents)
+    graph.refresh_schema()
+
+
+class TestSpannerGraphQAChain:
+
+    @pytest.fixture(scope="module")
+    def setup_db_load_data(self):
+        graph = get_spanner_graph()
+        load_data(graph)
+        yield graph
+        # teardown
+        print(graph.get_schema)
+        graph.cleanup()
+
+    @pytest.fixture
+    def chain(self, setup_db_load_data):
+        graph = setup_db_load_data
+        return SpannerGraphQAChain.from_llm(
+            get_llm(),
+            graph=graph,
+            verbose=True,
+            return_intermediate_steps=True,
+            allow_dangerous_requests=True,
+        )
+
+    @pytest.fixture
+    def chain_without_opt_in(self, setup_db_load_data):
+        graph = setup_db_load_data
+        return SpannerGraphQAChain.from_llm(
+            get_llm(),
+            graph=graph,
+            verbose=True,
+            return_intermediate_steps=True,
+        )
+
+    def test_spanner_graph_qa_chain_1(self, chain):
+        question = "Where does Elias Thorne's sibling live?"
+        response = chain.invoke("query=" + question)
+        print(response)
+
+        answer = response["result"]
+        assert (
```

Comment:
just curious:

  is it a standard approach to evaluate answer by embedding?
  would it potentially be flaky?

Reply:
Yes, since we are checking with an LLM generated response which could be slightly different, check semantic similarity using embedding is often used (to reduce flakiness).